paasta_tools/setup_prometheus_adapter_config.py

```diff
@@ -305,9 +305,12 @@ def create_instance_active_requests_scaling_rule(
         ) by (kube_deployment))
     """
     load_per_instance = f"""
-        avg(
-            paasta_instance:envoy_cluster__egress_cluster_upstream_rq_active{{{worker_filter_terms}}}
-        ) by (kube_pod, kube_deployment)
+        label_replace(
+            avg(
+                paasta_instance:envoy_cluster__egress_cluster_upstream_rq_active{{{worker_filter_terms}}}
+            ),
+            "kube_deployment", {deployment_name}, "", "(.*)"
```

Comment:
not sure if this matters, but
```suggestion
            "kube_deployment", {deployment_name}, "", ""
```
might make it clearer that we're just trying to add a label to the final result set (and maybe be nicer to prometheus as there's no need to actually do any regex shenanigans?)

it might also be worth adding a comment above to the effect of 
```
# envoy-based metrics have no labels corresponding to the k8s resources that they
# front, but we can trivially add one in since our deployment names are of the form 
# {service_name}-{instance_name} - which are both things in `worker_filter_terms` so 
# it's safe to unconditionally add.
# this is necessary as otherwise the HPA/prometheus adapter does not know what these
# metrics are for.
```

Reply:
Makes sense. Will add in the comment and update the regex.